### PR TITLE
fixing account payment detail and create

### PR DESCRIPTION
**Pull request by jasperboyd**
This change is [Reviewable](https://reviewable.io/reviews/ten24/slatwall/4358)

### CODE REVIEW

File: model/entity/AccountPaymentMethod.cfc [hunk 1]
```diff
@@ -229,7 +229,12 @@ component displayname="Account Payment Method" entityname="SlatwallAccountPaymen
 	}
 
 	public void function setupEncryptedProperties() {
-		if(getCreditCardType() != "Invalid" && !isNull(getPaymentMethod()) && !isNull(getPaymentMethod().getSaveAccountPaymentMethodEncryptFlag()) && getPaymentMethod().getSaveAccountPaymentMethodEncryptFlag()) {
+		if( len(getCreditCardNumber()) > 0
```
Comment (jasperboyd) [inline model/entity/AccountPaymentMethod.cfc]: @sumitverma does this constitute as a change to how we handle credit cards?

File: model/validation/AccountPaymentMethod.json [hunk 2]
```diff
@@ -23,6 +23,6 @@
 		"paymentTransactions":	[{"contexts":"delete","maxCollection":0}]
 	},
 	"populatedPropertyValidation":{
-	    "billingAddress":[{"validate":"full"}]
+	    "billingAddress":[{"conditions":"paymentTypeCreditCard","validate":"full"}]
```
Comment (gregmoser) [inline model/validation/AccountPaymentMethod.json]: This additional validation is no good.  If a billing address is submitted with any type of payment, then it needs to be validated in full.

**Comment by migueltarga** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ten24/slatwall/4358)*

File: model/validation/AccountPaymentMethod.json (hunk 2 above)
Comment (jasperboyd) [inline model/validation/AccountPaymentMethod.json]: @gregmoser I'll flip the logic so it will only not check billing address if it's a gift card

Comment (gregmoser) [inline model/validation/AccountPaymentMethod.json]: I think that the point is... it shouldn't be checking billing address as it stands right now already... unless you pass a billingAddress form value.

**Pull request merged**